tests/test_item_resource.py

```diff
@@ -0,0 +1,196 @@
+    def test_create_item_with_empty_strings(self):
+        new_item = {
+            "title": "",
+            "description": "this is my first title"
+        }
+        response = self.client.post(
+            '/api/v1/bucketlists/{}/items'.format(self.bucketlist_with_items_id),
+            data=json.dumps(new_item),
+            headers=self.headers)
+        print(response.data)
+        self.assertTrue(b'empty strings not allowed' in response.data)
+
+    def test_cant_create_item_without_authentication(self):
+        new_item = {
+            "title": "new item title",
+            "description": "this is my first title"
+        }
+        no_token = self.headers
+        no_token['Authorization'] = ""
+        response = self.client.post(
+            '/api/v1/bucketlists/{}/items'.format(self.bucketlist_with_items_id),
+            data=json.dumps(new_item),
+            headers=no_token)
+        self.assertTrue(b'Authorization Required' in response.data)
+
+
+    #update item
+    def test_item_updated_successfully_message(self):
+        updates = {
+            "title": "not item one anymore",
+            "description": "this was changed"
+        }
+        response = self.client.put(
+            '/api/v1/bucketlists/{}/items/{}'.format(self.bucketlist_with_items_id, self.item_one_id),
+            data=json.dumps(updates),
+            headers=self.headers)
+        self.assertTrue(b'item updated successfully' in response.data)
+
+    def test_message_on_update_with_missing_parameter(self):
+        updates = {
+            "description": "descriptive"
+        }
+        response = self.client.put(
+            '/api/v1/bucketlists/{}/items/{}'.format(self.bucketlist_with_items_id, self.item_one_id),
+            data=json.dumps(updates),
+            headers=self.headers)
+        self.assertTrue(b'Missing required parameter' in response.data)
+
+    def test_message_on_update_with_empty_parameters(self):
+        updates = {
+            "title": "titular",
+            "description": ""
+        }
+        response = self.client.put(
+            '/api/v1/bucketlists/{}/items/{}'.format(self.bucketlist_with_items_id, self.item_one_id),
+            data=json.dumps(updates),
+            headers=self.headers)
+        self.assertTrue(b'empty strings not allowed' in response.data)
+
+    #delete item
+    def test_item_successfully_deleted(self):
+        response = self.client.delete(
+            '/api/v1/bucketlists/{}/items/{}'.format(
+                self.bucketlist_with_items_id, self.item_one_id),
+            headers=self.headers)
+        self.assertTrue(b'item deleted successfully' in response.data)
+
+    def test_message_for_delete_non_existing_id(self):
+        response = self.client.delete(
+            '/api/v1/bucketlists/{}/items/{}'.format(
+                self.bucketlist_with_items_id, 40000),
+            headers=self.headers)
+        self.assertTrue(b'item does not exist' in response.data)
+
+    
```

Comment:
Suggestion for the tests; Create `user1` and use his/her token and try to delete items in `user2`'s bucketlist